features/joinroom/impl/src/main/kotlin/io/element/android/features/joinroom/impl/JoinRoomPresenter.kt

```diff
@@ -113,15 +113,21 @@ class JoinRoomPresenter @AssistedInject constructor(
                             // Workaround to get info about the sender for banned rooms
                             // TODO re-do this once we have a better API in the SDK
                             val preview = matrixClient.getRoomPreview(roomIdOrAlias, serverNames)
-                            val membershipDetalis = preview.getOrNull()?.membershipDetails()?.getOrNull()
-                            membershipDetalis?.senderMember to membershipDetalis?.currentUserMember?.membershipChangeReason
+                            val membershipDetails = preview.getOrNull()?.membershipDetails()?.getOrNull()
+                            membershipDetails?.senderMember to membershipDetails?.currentUserMember?.membershipChangeReason
                         }
                         CurrentUserMembership.INVITED -> {
                             roomInfo.get().inviter to null
                         }
                         else -> null to null
                     }
-                    value = roomInfo.get().toContentState(sender, reason)
+                    val preview = matrixClient.getRoomPreview(roomIdOrAlias, serverNames)
```

Comment:
If the user is banned we have 2 get room preview calls. Maybe the newly added one could be before this block and reuse it for the banned state?

Reply:
Ah true, good catch. Updated in 10d79ab5d4734f7404e3d506ae5c08c73d4f95d9